### PR TITLE
Upgrade to support ES 5.x, refs #10847

### PR DESCRIPTION
**Pull request by jraddaoui**
Upgrade Elasticsearch support from 1.x to 5.x. Main changes:

- Update Elastica to v5.2.1 and add new required deps.: Psr-fig/log and
  elasticsearch-php. The later requires PHP >= v5.6.6, so that should be
  the min. PHP version for AtoM after this changes.
- Fix ES status check on install process.
- Fix ES error display (ElasticsearchException is deprecated).
- Update ES index mappings:
  - String fields can now be keyword (not analyzed) or text (analyzed) and
    the not analyzed/analyzed value is not needed in the index property.
  - Remove include_in_all inside multi fields.
  - Change usage of index_analyzer to analyzer.
  - Increase total fields limit to 2000.
- Update ES/Elastica queries:
  - Use setSize instead of setLimit.
  - Use setSource instead of setFields.
  - Filters are replaced by queries, which can be applied as filters using the
    setPostFilter method from the queries. There is no query for the missing
    and not filters, they have been replaced by using a must_not clause in a
    bool query.
  - Replace facets by aggregations: mostly the same with some simplifications
    in the populate method and the template partial. It removes the partially
    introduced hability of selecting multiple values. Some changes required in
    the unique records query for the language facet as the language query is
    no longer added as a filter. Drafts are now filtered in the query so there is
    no need to do it in the aggs.
- Fix empty resultset creation, a third param with the results is now needed.
- Fix search-box autocomplete display, the flat option is no longer needed
  in QubitHelper get_search_i18n.

I have not renamed the HTML classes and ids related to the facets to avoid
making a lot of changes in external customizations after the upgrade.

### CODE REVIEW

**Comment by sevein** [comment]
Looking great so far! 👏

File: apps/qubit/modules/informationobject/templates/_cardViewResults.php [hunk 1]
```diff
@@ -11,15 +11,15 @@
     <?php endif; ?>
 
       <a href="<?php echo url_for(array('module' => 'informationobject', 'slug' => $doc['slug'])) ?>">
-        <?php if (!empty($doc['digitalObject']['thumbnailPath'])
+        <?php if (isset($doc['digitalObject']) && !empty($doc['digitalObject']['thumbnailPath'])
```
Comment (sevein) [inline apps/qubit/modules/informationobject/templates/_cardViewResults.php]: Are you sure this change is necessary? empty() not enough? This seems to work for me:

```php
<?php

error_reporting(E_ALL);
ini_set('display_errors', '1');

$a = array();
empty($a['foo']['bar']);

$a = NULL;
empty($a['foo']['bar']);

Comment (jraddaoui) [reply apps/qubit/modules/informationobject/templates/_cardViewResults.php]: If I leave it as it was I get the following notice, maybe it has something to be with how the sfOutputEscaperArrayDecorator works.:

```[Mon Jul 03 17:18:58.160718 2017] [:error] [pid 30542] [client 192.168.56.1:55952] PHP Notice:  Undefined index: digitalObject in /home/radda/www/atom-2/vendor/symfony/lib/escaper/sfOutputEscaperArrayDecorator.class.php on line 119, referer: http://192.168.56.128:81/atom-2/qubit_dev.php/informationobject/browse
[Mon Jul 03 17:18:58.160724 2017] [:error] [pid 30542] [client 192.168.56.1:55952] PHP Stack trace:, referer: http://192.168.56.128:81/atom-2/qubit_dev.php/informationobject/browse
[Mon Jul 03 17:18:58.160728 2017] [:error] [pid 30542] [client 192.168.56.1:55952] PHP   1. {main}() /home/radda/www/atom-2/qubit_dev.php:0, referer: http://192.168.56.128:81/atom-2/qubit_dev.php/informationobject/browse
[Mon Jul 03 17:18:58.160731 2017] [:error] [pid 30542] [client 192.168.56.1:55952] PHP   2. sfContext->dispatch() /home/radda/www/atom-2/qubit_dev.php:21, referer: http://192.168.56.128:81/atom-2/qubit_dev.php/informationobject/browse
[Mon Jul 03 17:18:58.160734 2017] [:error] [pid 30542] [client 192.168.56.1:55952] PHP   3. sfFrontWebController->dispatch() /home/radda/www/atom-2/vendor/symfony/lib/util/sfContext.class.php:170, referer: http://192.168.56.128:81/atom-2/qubit_dev.php/informationobject/browse
[Mon Jul 03 17:18:58.160741 2017] [:error] [pid 30542] [client 192.168.56.1:55952] PHP   4. sfController->forward() /home/radda/www/atom-2/vendor/symfony/lib/controller/sfFrontWebController.class.php:48, referer: http://192.168.56.128:81/atom-2/qubit_dev.php/informationobject/browse
[Mon Jul 03 17:18:58.160744 2017] [:error] [pid 30542] [client 192.168.56.1:55952] PHP   5. sfFilterChain->execute() /home/radda/www/atom-2/vendor/symfony/lib/controller/sfController.class.php:238, referer: http://192.168.56.128:81/atom-2/qubit_dev.php/informationobject/browse
[Mon Jul 03 17:18:58.160747 2017] [:error] [pid 30542] [client 192.168.56.1:55952] PHP   6. QubitTransactionFilter->execute() /home/radda/www/atom-2/vendor/symfony/lib/filter/sfFilterChain.class.php:53, referer: http://192.168.56.128:81/atom-2/qubit_dev.php/informationobject/browse
[Mon Jul 03 17:18:58.160750 2017] [:error] [pid 30542] [client 192.168.56.1:55952] PHP   7. sfFilterChain->execute() /home/radda/www/atom-2/lib/filter/QubitTransactionFilter.class.php:48, referer: http://192.168.56.128:81/atom-2/qubit_dev.php/informationobject/browse
[Mon Jul 03 17:18:58.160753 2017] [:error] [pid 30542] [client 192.168.56.1:55952] PHP   8. sfHistoryPluginFilter->execute() /home/radda/www/atom-2/vendor/symfony/lib/filter/sfFilterChain.class.php:53, referer: http://192.168.56.128:81/atom-2/qubit_dev.php/informationobject/browse
[Mon Jul 03 17:18:58.160756 2017] [:error] [pid 30542] [client 192.168.56.1:55952] PHP   9. sfFilterChain->execute() /home/radda/www/atom-2/plugins/sfHistoryPlugin/lib/sfHistoryPluginFilter.class.php:18, referer: http://192.168.56.128:81/atom-2/qubit_dev.php/informationobject/browse
[Mon Jul 03 17:18:58.160759 2017] [:error] [pid 30542] [client 192.168.56.1:55952] PHP  10. sfRenderingFilter->execute() /home/radda/www/atom-2/vendor/symfony/lib/filter/sfFilterChain.class.php:53, referer: http://192.168.56.128:81/atom-2/qubit_dev.php/informationobject/browse
[Mon Jul 03 17:18:58.160762 2017] [:error] [pid 30542] [client 192.168.56.1:55952] PHP  11. sfFilterChain->execute() /home/radda/www/atom-2/vendor/symfony/lib/filter/sfRenderingFilter.class.php:33, referer: http://192.168.56.128:81/atom-2/qubit_dev.php/informationobject/browse
[Mon Jul 03 17:18:58.160765 2017] [:error] [pid 30542] [client 192.168.56.1:55952] PHP  12. QubitSettingsFilter->execute() /home/radda/www/atom-2/vendor/symfony/lib/filter/sfFilterChain.class.php:53, referer: http://192.168.56.128:81/atom-2/qubit_dev.php/informationobject/browse
[Mon Jul 03 17:18:58.160767 2017] [:error] [pid 30542] [client 192.168.56.1:55952] PHP  13. sfFilterChain->execute() /home/radda/www/atom-2/lib/filter/QubitSettingsFilter.class.php:64, referer: http://192.168.56.128:81/atom-2/qubit_dev.php/informationobject/browse
[Mon Jul 03 17:18:58.160770 2017] [:error] [pid 30542] [client 192.168.56.1:55952] PHP  14. QubitLimitIpFilter->execute() /home/radda/www/atom-2/vendor/symfony/lib/filter/sfFilterChain.class.php:53, referer: http://192.168.56.128:81/atom-2/qubit_dev.php/informationobject/browse
[Mon Jul 03 17:18:58.160772 2017] [:error] [pid 30542] [client 192.168.56.1:55952] PHP  15. sfFilterChain->execute() /home/radda/www/atom-2/lib/filter/QubitLimitIp.class.php:39, referer: http://192.168.56.128:81/atom-2/qubit_dev.php/informationobject/browse
[Mon Jul 03 17:18:58.160775 2017] [:error] [pid 30542] [client 192.168.56.1:55952] PHP  16. QubitSslRequirementFilter->execute() /home/radda/www/atom-2/vendor/symfony/lib/filter/sfFilterChain.class.php:53, referer: http://192.168.56.128:81/atom-2/qubit_dev.php/informationobject/browse
[Mon Jul 03 17:18:58.160778 2017] [:error] [pid 30542] [client 192.168.56.1:55952] PHP  17. sfFilterChain->execute() /home/radda/www/atom-2/lib/filter/QubitSslRequirementFilter.class.php:35, referer: http://192.168.56.128:81/atom-2/qubit_dev.php/informationobject/browse
[Mon Jul 03 17:18:58.160780 2017] [:error] [pid 30542] [client 192.168.56.1:55952] PHP  18. QubitMeta->execute() /home/radda/www/atom-2/vendor/symfony/lib/filter/sfFilterChain.class.php:53, referer: http://192.168.56.128:81/atom-2/qubit_dev.php/informationobject/browse
[Mon Jul 03 17:18:58.160785 2017] [:error] [pid 30542] [client 192.168.56.1:55952] PHP  19. sfFilterChain->execute() /home/radda/www/atom-2/lib/filter/QubitMeta.class.php:31, referer: http://192.168.56.128:81/atom-2/qubit_dev.php/informationobject/browse
[Mon Jul 03 17:18:58.160787 2017] [:error] [pid 30542] [client 192.168.56.1:55952] PHP  20. QubitLimitResults->execute() /home/radda/www/atom-2/vendor/symfony/lib/filter/sfFilterChain.class.php:53, referer: http://192.168.56.128:81/atom-2/qubit_dev.php/informationobject/browse
[Mon Jul 03 17:18:58.160790 2017] [:error] [pid 30542] [client 192.168.56.1:55952] PHP  21. sfFilterChain->execute() /home/radda/www/atom-2/lib/filter/QubitLimitResults.class.php:34, referer: http://192.168.56.128:81/atom-2/qubit_dev.php/informationobject/browse
[Mon Jul 03 17:18:58.160793 2017] [:error] [pid 30542] [client 192.168.56.1:55952] PHP  22. sfExecutionFilter->execute() /home/radda/www/atom-2/vendor/symfony/lib/filter/sfFilterChain.class.php:53, referer: http://192.168.56.128:81/atom-2/qubit_dev.php/informationobject/browse
[Mon Jul 03 17:18:58.160795 2017] [:error] [pid 30542] [client 192.168.56.1:55952] PHP  23. sfExecutionFilter->handleView() /home/radda/www/atom-2/vendor/symfony/lib/filter/sfExecutionFilter.class.php:47, referer: http://192.168.56.128:81/atom-2/qubit_dev.php/informationobject/browse
[Mon Jul 03 17:18:58.160798 2017] [:error] [pid 30542] [client 192.168.56.1:55952] PHP  24. sfExecutionFilter->executeView() /home/radda/www/atom-2/vendor/symfony/lib/filter/sfExecutionFilter.class.php:116, referer: http://192.168.56.128:81/atom-2/qubit_dev.php/informationobject/browse
[Mon Jul 03 17:18:58.160801 2017] [:error] [pid 30542] [client 192.168.56.1:55952] PHP  25. sfPHPView->render() /home/radda/www/atom-2/vendor/symfony/lib/filter/sfExecutionFilter.class.php:155, referer: http://192.168.56.128:81/atom-2/qubit_dev.php/informationobject/browse
[Mon Jul 03 17:18:58.160804 2017] [:error] [pid 30542] [client 192.168.56.1:55952] PHP  26. sfPHPView->renderFile() /home/radda/www/atom-2/vendor/symfony/lib/view/sfPHPView.class.php:185, referer: http://192.168.56.128:81/atom-2/qubit_dev.php/informationobject/browse
[Mon Jul 03 17:18:58.160806 2017] [:error] [pid 30542] [client 192.168.56.1:55952] PHP  27. require() /home/radda/www/atom-2/vendor/symfony/lib/view/sfPHPView.class.php:75, referer: http://192.168.56.128:81/atom-2/qubit_dev.php/informationobject/browse
[Mon Jul 03 17:18:58.160809 2017] [:error] [pid 30542] [client 192.168.56.1:55952] PHP  28. get_partial() /home/radda/www/atom-2/apps/qubit/modules/informationobject/templates/browseSuccess.php:338, referer: http://192.168.56.128:81/atom-2/qubit_dev.php/informationobject/browse
[Mon Jul 03 17:18:58.160812 2017] [:error] [pid 30542] [client 192.168.56.1:55952] PHP  29. sfPartialView->render() /home/radda/www/atom-2/vendor/symfony/lib/helper/PartialHelper.php:220, referer: http://192.168.56.128:81/atom-2/qubit_dev.php/informationobject/browse
[Mon Jul 03 17:18:58.160814 2017] [:error] [pid 30542] [client 192.168.56.1:55952] PHP  30. sfPHPView->renderFile() /home/radda/www/atom-2/vendor/symfony/lib/view/sfPartialView.class.php:124, referer: http://192.168.56.128:81/atom-2/qubit_dev.php/informationobject/browse
[Mon Jul 03 17:18:58.160817 2017] [:error] [pid 30542] [client 192.168.56.1:55952] PHP  31. require() /home/radda/www/atom-2/vendor/symfony/lib/view/sfPHPView.class.php:75, referer: http://192.168.56.128:81/atom-2/qubit_dev.php/informationobject/browse
[Mon Jul 03 17:18:58.160820 2017] [:error] [pid 30542] [client 192.168.56.1:55952] PHP  32. sfOutputEscaperArrayDecorator->offsetGet() /home/radda/www/atom-2/vendor/symfony/lib/view/sfPHPView.class.php:14, referer: http://192.168.56.128:81/atom-2/qubit_dev.php/informationobject/browse
```

Comment (sevein) [reply apps/qubit/modules/informationobject/templates/_cardViewResults.php]: Ugh, right, `sfOutputEscaperArrayDecorator`... that's sad but nothing we can do!

File: apps/qubit/modules/user/actions/clipboardAction.class.php [hunk 2]
```diff
@@ -41,7 +41,7 @@ public function execute($request)
 
     if (!isset($allSlugs[$this->entityType]) || !count($allSlugs[$this->entityType]))
     {
-      $resultSet = new \Elastica\ResultSet(new Elastica\Response(null), new Elastica\Query);
+      $resultSet = new \Elastica\ResultSet(new Elastica\Response(null), new Elastica\Query, array());
```
Comment (sevein) [inline apps/qubit/modules/user/actions/clipboardAction.class.php:44]: 😅

File: apps/qubit/modules/search/templates/_aggregation.php [hunk 3]
```diff
@@ -0,0 +1,36 @@
+<?php if (!isset($aggs[$name]) || (!isset($filters[$name]) && count($aggs[$name]) < 2)) return ?>
```
Comment (MikeFE) [inline apps/qubit/modules/search/templates/_aggregation.php:1]: What is this doing? Does return here just make it not render the partial?

File: apps/qubit/modules/search/templates/_aggregation.php [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+<?php $openned = (isset($sf_request->$name) || (isset($open) && $open && 0 < count($aggs[$name]))) ?>
```
Comment (MikeFE) [inline apps/qubit/modules/search/templates/_aggregation.php:2]: opened

File: apps/qubit/modules/search/templates/_aggregation.php (hunk 3 above)
Comment (jraddaoui) [reply apps/qubit/modules/search/templates/_aggregation.php:1]: Sorry Mike, I forgot about this. Yes, it works like in an action.

File: apps/qubit/modules/search/templates/_aggregation.php (hunk 4 above)
Comment (jraddaoui) [reply apps/qubit/modules/search/templates/_aggregation.php:2]: Thanks! Will fix.

**Comment by jraddaoui** [comment]
Merged in https://github.com/artefactual/atom/commit/eabbac520d765a4a903e01f0ef77bc068d71b7a7

**Pull request closed without merging**